tests/common/aws_test_runner.c

```diff
@@ -266,6 +275,107 @@ static void RunTests( void )
     #endif
 }
 /*-----------------------------------------------------------*/
+#if defined( AWS_TEST_RUNNER_ENABLE_CLI_INPUT ) && ( AWS_TEST_RUNNER_ENABLE_CLI_INPUT == 1 )
+
+/* Global flag that represents whether a "start" command is received
+ * on the serial console. */
+    static BaseType_t xReceivedCommand = pdFALSE;
+
+/**
+ * @brief The command handler for the "start" command recognized by the
+ * serial console of the FreeRTOS test runner. On receiving the command,
+ * the test runner begins executing tests on the device.
+ *
+ * @return Returns pdFALSE to indicate completion of execution of tests.
+ */
+    static BaseType_t prvRunTestsCommand( int8_t * pcWriteBuffer,
+                                          size_t xWriteBufferLen,
+                                          const int8_t * pcCommandString )
+    {
+        ( void ) pcWriteBuffer;
+        ( void ) xWriteBufferLen;
+        ( void ) pcCommandString;
+
+        /* Set flag to indicate reception of command from serial console to start */
+        /* executing tests. */
+        xReceivedCommand = pdTRUE;
+
+        UNITY_BEGIN();
+
+        RunTests();
```

Comment:
Needs to handle `UNITY_END()` ?

Reply:
`UNITY_END` is called after the console CLI exists.

Reply:
Have moved all the test logic (of setup, executing tests and tear down) to common function, which is called from the handler now.